src/main/drivers/flash_impl.h

```diff
@@ -28,8 +28,23 @@
 
 struct flashVTable_s;
 
+typedef enum {
+    FLASHIO_NONE = 0,
+    FLASHIO_SPI,
+    FLASHIO_QUADSPI
+} flashDeviceIoMode_e;
+
+typedef struct flashDeviceIO_s {
+    union {
```

Comment:
I don't think we are using nested type definitions so far. The question is: Do we want to start this now?

Reply:
Maybe quadSpi should be added into busDevice_t...
There may be some cases when other drivers expect only I2C/SPI, probably quite easy to fix.

Reply:
@mikeller If you are referring to `union` inside `struct`, it has been used in `busDevice_t`.

@ledvinap I believe that the design was selected for least intrusiveness. @hydra do you have anything else to say?

Reply:
quadspi does not follow the pattern of spi or i2c, at all.  with qspi you create commands and then data, as opposed to just sending random data over the wire.
 the format of the commands and data is also configurable.

refer to the stm32 quadspi section in the reference manual.

I initially starting by adding quadspi to busdevice but it just doesn't make sense due to the way you interact with the hardware